resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/DriverServiceFeatureStep.scala

```diff
@@ -39,14 +39,15 @@ private[spark] class DriverServiceFeatureStep(
       "managed via a Kubernetes service.")
 
   private val preferredServiceName = s"${kubernetesConf.resourceNamePrefix}$DRIVER_SVC_POSTFIX"
-  private val resolvedServiceName = if (preferredServiceName.length <= MAX_SERVICE_NAME_LENGTH) {
+  private val resolvedServiceName = if (preferredServiceName.length <= MAX_SERVICE_NAME_LENGTH
+    && Character.isLetter(preferredServiceName.charAt(0))) {
```

Comment:
Is the svc name valid according to `com.google.common.net.InternetDomainName.isValid` as mentioned in the other PR?

Reply:
> Is the svc name valid in all cases according to `com.google.common.net.InternetDomainName.isValid` as mentioned in the other PR? As it is now it seems valid according to the spec: https://en.wikipedia.org/wiki/Domain_name.

Now,  fabric8 API has some limitations，
```java
 public class Service implements HasMetadata {
    @NotNull
    @JsonProperty("apiVersion")
    private String apiVersion = "v1";
    @NotNull
    @JsonProperty("kind")
    private String kind = "Service";
    @JsonProperty("metadata")
    @Valid
    @CheckObjectMeta(
        regexp = "^[a-z]([-a-z0-9]*[a-z0-9])?$",
        max = 63
    )
```

`    @CheckObjectMeta(
        regexp = "^[a-z]([-a-z0-9]*[a-z0-9])?$",
        max = 63
    )
`

Reply:
Yeah I saw that when digging into it. It should be ok from what I read: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/ "Assume a Service named foo in the Kubernetes namespace bar. A Pod running in namespace bar can look up this service by simply doing a DNS query for foo. A Pod running in namespace quux can look up this service by doing a DNS query for foo.bar." so it matches the domain label, that is why there are these limitations.

Reply:
I don't understand the purpose of  `Character.isLetter` here, we've already filtered this down to only allow `[a-z0-9]`